app/code/community/Ebizmarts/MailChimp/Helper/Data.php

```diff
@@ -2282,9 +2282,19 @@ public function addResendFilter($collection, $magentoStoreId, $itemType)
                     $this->logError($this->__('The item type sent in the filter does not match any of the available options.'));
             }
             if ($resendTurn) {
-                $collection->addFieldToFilter('entity_id', array('lteq' => $lastItemSent));
+                if($itemType == Ebizmarts_MailChimp_Model_Config::IS_PROMO_CODE) {
+                    $collection->addFieldToFilter('coupon_id', array('lteq' => $lastItemSent));
+                }
+                else {
+                    $collection->addFieldToFilter('entity_id', array('lteq' => $lastItemSent));
+                }
             } else {
-                $collection->addFieldToFilter('entity_id', array('gt' => $lastItemSent));
+                if($itemType == Ebizmarts_MailChimp_Model_Config::IS_PROMO_CODE) {
```

Comment:
Please change this to stop repeating the same condition multiple times.
Adding a variable and setting it in the switch could do the same.